src/main/java/meteordevelopment/meteorclient/systems/modules/combat/CrystalAura.java

```diff
@@ -1189,29 +1203,40 @@ private float getDamageToTargets(Vec3d vec3d, BlockPos obsidianPos, boolean brea
 
     @Override
     public String getInfoString() {
-        return bestTarget != null && bestTargetTimer > 0 ? bestTarget.getGameProfile().getName() : null;
+        return bestTarget != null && bestTargetTimer > 0 ? bestTarget.getName().toString() : null;
     }
 
     private void findTargets() {
         targets.clear();
 
-        // Players
-        for (PlayerEntity player : mc.world.getPlayers()) {
-            if (player.getAbilities().creativeMode || player == mc.player) continue;
-            if (!player.isAlive() || !Friends.get().shouldAttack(player)) continue;
-            if (player.squaredDistanceTo(mc.player) > Math.pow(targetRange.get(), 2)) continue;
-
-            if (ignoreNakeds.get()) {
-                if (player.getOffHandStack().isEmpty()
-                    && player.getMainHandStack().isEmpty()
-                    && player.getInventory().armor.get(0).isEmpty()
-                    && player.getInventory().armor.get(1).isEmpty()
-                    && player.getInventory().armor.get(2).isEmpty()
-                    && player.getInventory().armor.get(3).isEmpty()
-                ) continue;
+        // Living Entities
+        for (Entity entity : mc.world.getEntities()) {
+            // Ignore non-living
+            if (!(entity instanceof LivingEntity)) continue;
+
+            // Player
+            if (entity instanceof PlayerEntity player) {
+                if (player.getAbilities().creativeMode || entity == mc.player) continue;
+                if (!player.isAlive() || !Friends.get().shouldAttack(player)) continue;
+
+                if (ignoreNakeds.get()) {
+                    if (player.getOffHandStack().isEmpty()
+                        && player.getMainHandStack().isEmpty()
+                        && player.getInventory().armor.get(0).isEmpty()
+                        && player.getInventory().armor.get(1).isEmpty()
+                        && player.getInventory().armor.get(2).isEmpty()
+                        && player.getInventory().armor.get(3).isEmpty()
+                    ) continue;
+                }
             }
 
-            targets.add(player);
+            // Animals, water animals, monsters, bats, misc
+            if (!(entities.get().contains(entity.getType()))) continue;
+
+            // Close enough to damage
+            if (entity.squaredDistanceTo(mc.player) > Math.pow(targetRange.get(), 2)) continue;
+
+            targets.add((LivingEntity) entity);
```

Comment:
instead of casting here, use a pattern matching `instanceof` on line 1215

Reply:
like this? 
```java
targets.add(entity instanceof LivingEntity);
```

that produces a boolean, which isn't what we want here. I'm not sure I understand.

Reply:
```java
if (entity instanceof LivingEntity livingEntity) {
    //code
}
```